internal/validations/clusterlogforwarder/validate_http_content_type_headers.go

```diff
@@ -0,0 +1,35 @@
+package clusterlogforwarder
+
+import (
+	"fmt"
+	log "github.com/ViaQ/logerr/v2/log/static"
+	"github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	"reflect"
+	"strings"
+)
+
+var validContentTypes = map[string]string{
+	"application/json":     "json",
+	"application/x-ndjson": "ndjson",
+}
+
+// validateHttpContentTypeHeaders will validate Content-Type header in Http Output
+// valid content-type are: "application/json" and "application/x-ndjson"
+// was introduced in https://github.com/openshift/cluster-logging-operator/pull/1924
+// for https://issues.redhat.com/browse/LOG-3784
+func validateHttpContentTypeHeaders(clf v1.ClusterLogForwarder) error {
+	for i, output := range clf.Spec.Outputs {
+		_, output := i, output // Don't bind range variable.
+		if output.Type == v1.OutputTypeHttp && output.Http != nil && len(output.Http.Headers) != 0 && output.Http.Headers["Content-Type"] != "" {
```

Comment:
I don't think you need to check the length.  I believe this will achieve the same:
```
if value, found := output.Http.Headers["Content-Type"]; found && value  != "" {
  //do something 
}
```